lib/Sema/TypeCheckType.cpp

```diff
@@ -1996,7 +1996,19 @@ Type TypeResolver::resolveAttributedType(TypeAttributes &attrs,
       hasFunctionAttr = true;
       break;
     }
-
+  
+  // The top level repr may point to a typealias, so lets resolve the repr
+  // and get the underlying type repr
+  if (auto CITR = dyn_cast_or_null<ComponentIdentTypeRepr>(repr)) {
```

Comment:
dyn_cast_or_null doesn’t mean what you think, it doesn’t allow nullptr to be passed to it as an argument

Reply:
Ah ok. Seems like its because the underlying repr might be `null`. I have added a check and it fixes the crash.